.github/workflows/main.yml

```diff
@@ -1,225 +1,187 @@
         with:
-          path: ${{ github.workspace }}/venv
-          key: venv-${{ github.ref }}-${{ hashFiles('requirements.txt') }}
-
-  format-py:
-    needs: install-py
-    runs-on: ubuntu-latest
+          path: ~/.npm
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
 
-    steps:
-      - uses: actions/checkout@v2
-      - uses: "actions/setup-python@v2"
+      - name: Install Node.js
+        uses: actions/setup-node@v2
         with:
-          python-version: "3.6"
-      - name: Use the cached venv
-        uses: actions/cache@v2
+          node-version: "16"
+
+      - name: Install node_modules
+        run: npm ci
+
+      - name: Build assets
+        run: npm run build
+
+      - name: Store assets
+        uses: actions/upload-artifact@v2
         with:
-          path: ${{ github.workspace }}/venv
-          key: venv-${{ github.ref }}-${{ hashFiles('requirements.txt') }}
-      - run: ls -lah ${{ github.workspace }}
-      - name: Check formatting
-        run: |
-          source ${{ github.workspace }}/venv/bin/activate
-          make format-py
+          name: node-assets
+          path: static/dist/js
 
-  lint-py:
-    needs: install-py
+  check:
+    needs: [build]
     runs-on: ubuntu-latest
 
     steps:
       - uses: actions/checkout@v2
       - uses: "actions/setup-python@v2"
         with:
-          python-version: "3.6"
-      - name: Use the cached venv
-        uses: actions/cache@v2
-        with:
-          path: ${{ github.workspace }}/venv
-          key: venv-${{ github.ref }}-${{ hashFiles('requirements.txt') }}
-      - name: Check linting
-        run: |
-          source ${{ github.workspace }}/venv/bin/activate
-          make lint-py
+          python-version: "3.8"
+          cache: pip
+          cache-dependency-path: requirements.*.txt
+      - uses: extractions/setup-just@aa5d15c144db4585980a44ebfdd2cf337c4f14cb  # v1.4.0
+      - name: Check formatting, linting and import sorting
+        run: just check
 
-  sort:
-    needs: install-py
+  check-js:
+    needs: [build]
     runs-on: ubuntu-latest
 
     steps:
       - uses: actions/checkout@v2
-      - uses: "actions/setup-python@v2"
-        with:
-          python-version: "3.6"
-      - name: Use the cached venv
-        uses: actions/cache@v2
-        with:
-          path: ${{ github.workspace }}/venv
-          key: venv-${{ github.ref }}-${{ hashFiles('requirements.txt') }}
-      - name: Check import sorting
-        run: |
-          source ${{ github.workspace }}/venv/bin/activate
-          make sort
+      - uses: extractions/setup-just@aa5d15c144db4585980a44ebfdd2cf337c4f14cb  # v1.4.0
+      - name: Check js linting
+        run: just check-js
 
   test-py:
-    needs: [format-py, lint-py, sort]
+    needs: [check]
```

Comment:
I've stopped gating tests on tooling checks in a number of repos since they often end up with a linear run time, so it turns out to be a tigher feedback loop to run them in parallel!

Reply:
This almost halves the workflow time, thanks for the suggestion :slightly_smiling_face: